docs/development.md

```diff
@@ -0,0 +1,34 @@
+# Development
+
+## Before you get started
+
+Please read the [contributing](../CONTRIBUTING.md) guidelines before getting started.
+
+## Guide
+
+Following steps will help you get ready with a local development stack to contribute and test your changes before publishing a PR.
+
+- Follow the [installation](install.md) steps. It is recommended to use the Helm to setup App Mesh controller.
+- Clone and checkout aws-app-mesh-controller-for-k8s.
+- Make code or configuration changes.
+- Build and push container image.
+```
+make image push
+```
+- Deploy latest changes to appmesh-controller
+```
+make deploy
+```
+- Use examples from [aws-app-mesh-examples](https://github.com/aws/aws-app-mesh-examples/tree/master/walkthroughs) to verify the controller behavior.
+
+## Updating App Mesh CRD
```

Comment:
Should `make go-fmt` also be added here?

Reply:
added. We should probably make it as pre-commit hooks.